camera-server: Add ability to support taking image in photo mode and photo in image mode

jnomikos opened the pull request:
One limitation I ran into with camera-server was the inability to be able to set the capability flags for [CAMERA_CAP_FLAGS_CAN_CAPTURE_IMAGE_IN_VIDEO_MODE](https://mavlink.io/en/messages/common.html#CAMERA_CAP_FLAGS_CAN_CAPTURE_IMAGE_IN_VIDEO_MODE) and [CAMERA_CAP_FLAGS_CAN_CAPTURE_VIDEO_IN_IMAGE_MODE](https://mavlink.io/en/messages/common.html#CAMERA_CAP_FLAGS_CAN_CAPTURE_VIDEO_IN_IMAGE_MODE). However, there are situations where a user may want to support this functionality. In an application I was working on, I wanted to be able to send these flags since the camera likely supported it.

This PR adds two new functions, support_image_in_video_mode and support_video_in_image_mode. A user can use these to enable the associated capability flags.

Please let me know if this is a good approach, or if there is a better way to support this functionality. Thank you!

julianoes commented:
Looks like you have to rebase both the proto branch and this one and point to it again.

jnomikos commented:
> Looks like you have to rebase both the proto branch and this one and point to it again.
Will do

julianoes commented:
I've merged the proto PR, and rebased (and squashed) everything here. Let's check CI and get it in.

jnomikos commented:
> I've merged the proto PR, and rebased (and squashed) everything here. Let's check CI and get it in.

Thank you. Apologies for not doing it myself. Got distracted with other things
